sawtooth/client.py

```diff
@@ -779,24 +779,34 @@ def get_transaction_status(self, txnid, timeout=30):
             request.get_method = lambda: 'HEAD'
             opener = urllib2.build_opener(self.proxy_handler)
             response = opener.open(request, timeout=timeout)
-            code = response.getcode()
-            response.close()
-            if code == 200:
-                return transaction.Status.committed
-            elif code == 302:
-                return transaction.Status.pending
-            else:
-                return transaction.Status.unknown
 
         except urllib2.HTTPError as err:
             LOGGER.error('peer operation on url %s failed with response: %d',
                          url, err.code)
+            if err.code == 302:
+                return transaction.Status.pending
+
+            raise MessageException('operation failed '
+                                   'with response: {0}'.format(err.code))
 
         except urllib2.URLError as err:
             LOGGER.error('peer operation on url %s failed: %s', url,
                          err.reason)
+            raise MessageException('operation failed: {0}'.format(err.reason))
+
+        except:
```

Comment:
Yes, I know this. But all of other functions catch all exceptions, such as [headrequest](https://github.com/hyperledger/sawtooth-core/blob/master/sawtooth/client.py#L83), [getmsg](https://github.com/hyperledger/sawtooth-core/blob/master/sawtooth/client.py#L113), [postmsg](https://github.com/hyperledger/sawtooth-core/blob/master/sawtooth/client.py#L159), [_geturl](https://github.com/hyperledger/sawtooth-core/blob/master/sawtooth/client.py#L859), [_posturl](https://github.com/hyperledger/sawtooth-core/blob/master/sawtooth/client.py#L919), I'm not sure is there any reason for these functions? or we should avoid bare excepts for these functions either?